### PR TITLE
Add Library flag to sharepoint CLI

### PR DESCRIPTION
**Pull request by ryanfkeepers**
Adds a flag to filter sharepoint details/restores by specific libraries (aka drives).  Also renames some vars across the CLI for consistency.

---

#### Does this PR need a docs update or release note?

- [x] :no_entry: No

#### Type of change

- [x] :sunflower: Feature

#### Issue(s)

* #2757

#### Test Plan

- [x] :muscle: Manual
- [x] :zap: Unit test

### CODE REVIEW

**Comment by ryanfkeepers** [comment]
Full library details:
```
cd ./corso/src; go run ./corso.go backup details sharepoint \
	--config-file /Users/keepers/go/src/github.com/alcionai/.corso_make.toml --no-stats --log-file /Users/keepers/go/src/github.com/alcionai/.corso.log --log-level warn  \
	--backup dc0c7272-eecb-4eb7-8463-c6d826adcda6 \
	
==================================================================================================
	WARNING!!! THIS IS A PRE-RELEASE COMMAND THAT MAY NOT FUNCTION PROPERLY, OR AT ALL
==================================================================================================

Logging to file: /Users/keepers/go/src/github.com/alcionai/.corso.log
Connecting to repository:      3s done
  ID            ItemName                                 Drive      ParentPath                                  Size    WebURL                         Created               Modified
  1a4fadb79d4f  database.db                              Rhine      Corso_Restore_10-Mar-2023_23-08-06          11 kB   https://10rqc2.sharepoint.com  2023-03-10T23:08:46Z  2023-03-10T23:08:46Z
  fc49767f4d45  FromBritannica.docx                      Rhine      Corso_Restore_10-Mar-2023_23-08-06          222 kB  https://10rqc2.sharepoint.com  2023-03-10T23:08:43Z  2023-03-10T23:08:45Z
  0ede745326f4  bathReplacement.xlsx                     Rhine      Corso_Restore_10-Mar-2023_23-08-06          15 kB   https://10rqc2.sharepoint.com  2023-03-10T23:08:44Z  2023-03-10T23:08:45Z
  7d0d4bcc7c1a  database.db                              Rhine                                                  11 kB   https://10rqc2.sharepoint.com  2023-03-10T19:47:53Z  2023-03-10T19:47:53Z
  8aadec8a08dd  FromBritannica.docx                      Rhine                                                  222 kB  https://10rqc2.sharepoint.com  2023-02-24T15:05:22Z  2023-02-24T15:06:15Z
  1b478ac19a8b  bathReplacement.xlsx                     Rhine                                                  15 kB   https://10rqc2.sharepoint.com  2023-02-24T16:41:49Z  2023-02-24T18:35:50Z
  9a9d3187c01c  bash-g2aeff47a0_640.png                  Documents                                              97 kB   https://10rqc2.sharepoint.com  2023-03-08T18:00:30Z  2023-03-08T18:00:30Z
  ec28fc0a5d3b  2023-02-20T06-57-11Z.log                 Documents                                              1.6 MB  https://10rqc2.sharepoint.com  2023-03-08T18:01:57Z  2023-03-08T18:01:57Z
  abcad45d78e6  charlesdeluvio-OI2Yqj7GDy0-unsplash.jpg  Documents                                              980 kB  https://10rqc2.sharepoint.com  2023-02-24T14:58:03Z  2023-02-24T14:58:03Z
  c8942c690fe5  database.db                              Documents                                              11 kB   https://10rqc2.sharepoint.com  2023-02-24T14:58:18Z  2023-02-24T14:58:18Z
  15279cdb8b4b  care.gif                                 Documents                                              128 kB  https://10rqc2.sharepoint.com  2023-03-08T18:00:11Z  2023-03-08T18:00:11Z
  e75ab41e8961  TestSlide.pptx                           Documents  Applied                                     9.6 MB  https://10rqc2.sharepoint.com  2023-02-28T01:09:15Z  2023-02-28T01:11:12Z
  379482215ec8  quote.docx                               Documents  Applied                                     21 kB   https://10rqc2.sharepoint.com  2023-02-24T14:58:43Z  2023-02-24T15:02:50Z
  fd89991b21ab  quote.docx                               Documents  Corso_Restore_10-Mar-2023_23-08-06/Applied  21 kB   https://10rqc2.sharepoint.com  2023-03-10T23:08:41Z  2023-03-10T23:08:41Z
  1ceb63f91877  TestSlide.pptx                           Documents  Corso_Restore_10-Mar-2023_23-08-06/Applied  9.6 MB  https://10rqc2.sharepoint.com  2023-03-10T23:08:23Z  2023-03-10T23:08:41Z
  4818689cbf0f  charlesdeluvio-OI2Yqj7GDy0-unsplash.jpg  Documents  Corso_Restore_10-Mar-2023_23-08-06          980 kB  https://10rqc2.sharepoint.com  2023-03-10T23:08:20Z  2023-03-10T23:08:21Z
  5d8aab092c12  bash-g2aeff47a0_640.png                  Documents  Corso_Restore_10-Mar-2023_23-08-06          97 kB   https://10rqc2.sharepoint.com  2023-03-10T23:08:18Z  2023-03-10T23:08:18Z
  da2fee5557f5  database.db                              Documents  Corso_Restore_10-Mar-2023_23-08-06          11 kB   https://10rqc2.sharepoint.com  2023-03-10T23:08:21Z  2023-03-10T23:08:22Z
  836f60770163  care.gif                                 Documents  Corso_Restore_10-Mar-2023_23-08-06          128 kB  https://10rqc2.sharepoint.com  2023-03-10T23:08:19Z  2023-03-10T23:08:19Z
  b2392f22e9c0  2023-02-20T06-57-11Z.log                 Documents  Corso_Restore_10-Mar-2023_23-08-06          1.6 MB  https://10rqc2.sharepoint.com  2023-03-10T23:08:15Z  2023-03-10T23:08:17Z
```

Selected by library:
```
cd ./corso/src; go run ./corso.go backup details sharepoint \
	--config-file /Users/keepers/go/src/github.com/alcionai/.corso_make.toml --no-stats --log-file /Users/keepers/go/src/github.com/alcionai/.corso.log --log-level warn  \
	--backup dc0c7272-eecb-4eb7-8463-c6d826adcda6 \
	--library rhine
==================================================================================================
	WARNING!!! THIS IS A PRE-RELEASE COMMAND THAT MAY NOT FUNCTION PROPERLY, OR AT ALL
==================================================================================================

Logging to file: /Users/keepers/go/src/github.com/alcionai/.corso.log
Connecting to repository:      3s done
  ID            ItemName              Drive  ParentPath                          Size    WebURL                         Created               Modified
  1a4fadb79d4f  database.db           Rhine  Corso_Restore_10-Mar-2023_23-08-06  11 kB   https://10rqc2.sharepoint.com  2023-03-10T23:08:46Z  2023-03-10T23:08:46Z
  fc49767f4d45  FromBritannica.docx   Rhine  Corso_Restore_10-Mar-2023_23-08-06  222 kB  https://10rqc2.sharepoint.com  2023-03-10T23:08:43Z  2023-03-10T23:08:45Z
  0ede745326f4  bathReplacement.xlsx  Rhine  Corso_Restore_10-Mar-2023_23-08-06  15 kB   https://10rqc2.sharepoint.com  2023-03-10T23:08:44Z  2023-03-10T23:08:45Z
  7d0d4bcc7c1a  database.db           Rhine                                      11 kB   https://10rqc2.sharepoint.com  2023-03-10T19:47:53Z  2023-03-10T19:47:53Z
  8aadec8a08dd  FromBritannica.docx   Rhine                                      222 kB  https://10rqc2.sharepoint.com  2023-02-24T15:05:22Z  2023-02-24T15:06:15Z
  1b478ac19a8b  bathReplacement.xlsx  Rhine                                      15 kB   https://10rqc2.sharepoint.com  2023-02-24T16:41:49Z  2023-02-24T18:35:50Z
 ```
  
 Also selectable by drive id:
 ```
 cd ./corso/src; go run ./corso.go backup details sharepoint \
	--config-file /Users/keepers/go/src/github.com/alcionai/.corso_make.toml --no-stats --log-file /Users/keepers/go/src/github.com/alcionai/.corso.log --log-level warn  \
	--backup dc0c7272-eecb-4eb7-8463-c6d826adcda6 \
	--library b!A2idAACqt02CM-LL19n1AzwUCFTQ6zBBguZ2c3VoaFM6tAN58odkS6Snk2lYkZN2
==================================================================================================
	WARNING!!! THIS IS A PRE-RELEASE COMMAND THAT MAY NOT FUNCTION PROPERLY, OR AT ALL
==================================================================================================

Logging to file: /Users/keepers/go/src/github.com/alcionai/.corso.log
Connecting to repository:      3s done
  ID            ItemName              Drive  ParentPath                          Size    WebURL                         Created               Modified
  1a4fadb79d4f  database.db           Rhine  Corso_Restore_10-Mar-2023_23-08-06  11 kB   https://10rqc2.sharepoint.com  2023-03-10T23:08:46Z  2023-03-10T23:08:46Z
  fc49767f4d45  FromBritannica.docx   Rhine  Corso_Restore_10-Mar-2023_23-08-06  222 kB  https://10rqc2.sharepoint.com  2023-03-10T23:08:43Z  2023-03-10T23:08:45Z
  0ede745326f4  bathReplacement.xlsx  Rhine  Corso_Restore_10-Mar-2023_23-08-06  15 kB   https://10rqc2.sharepoint.com  2023-03-10T23:08:44Z  2023-03-10T23:08:45Z
  7d0d4bcc7c1a  database.db           Rhine                                      11 kB   https://10rqc2.sharepoint.com  2023-03-10T19:47:53Z  2023-03-10T19:47:53Z
  8aadec8a08dd  FromBritannica.docx   Rhine                                      222 kB  https://10rqc2.sharepoint.com  2023-02-24T15:05:22Z  2023-02-24T15:06:15Z
  1b478ac19a8b  bathReplacement.xlsx  Rhine                                      15 kB   https://10rqc2.sharepoint.com  2023-02-24T16:41:49Z  2023-02-24T18:35:50Z
 ```

File: src/cli/utils/sharepoint.go [hunk 1]
```diff
@@ -167,6 +166,7 @@ func FilterSharePointRestoreInfoSelectors(
 	sel *selectors.SharePointRestore,
 	opts SharePointOpts,
 ) {
+	AddSharePointFilter(sel, opts.Library, sel.Library)
```
Comment (vkamra) [inline src/cli/utils/sharepoint.go:169]: I assume this doesn't do anything yet and that's a follow-up PR?

**Review by vkamra — APPROVED** [review]

File: src/cli/utils/sharepoint.go (hunk 1 above)
Comment (vkamra) [reply src/cli/utils/sharepoint.go:169]: Hmm - the unit test and test output you pasted indicates otherwise. Where does this get used?

Comment (vkamra) [reply src/cli/utils/sharepoint.go:169]: Ignore this conversation. I looked at the implementation of `AddSharePointFilter`

Comment (ryanfkeepers) [reply src/cli/utils/sharepoint.go:169]: nope, that's the whole logic.

**Pull request merged**